src/plugins/maps_ems/public/service_settings/service_settings.ts

```diff
@@ -108,25 +119,28 @@ export class ServiceSettings {
   async getTMSServices() {
     let allServices = [];
     if (this._hasTmsConfigured) {
-      //use tilemap.* settings from yml
-      const tmsService = _.cloneDeep(this.tmsOptionsFromConfig);
-      tmsService.id = TMS_IN_YML_ID;
-      tmsService.origin = ORIGIN.KIBANA_YML;
+      // use tilemap.* settings from yml
+      const tmsService: TmsLayer = {
+        ..._.cloneDeep(this.tmsOptionsFromConfig),
+        id: TMS_IN_YML_ID,
+        origin: ORIGIN.KIBANA_YML,
+      };
+
       allServices.push(tmsService);
     }
 
     if (this._mapConfig.includeElasticMapsService && !this._mapConfig.emsUrl) {
       const servicesFromManifest = await this._emsClient.getTMSServices();
-      const strippedServiceFromManifest = await Promise.all(
+      const strippedServiceFromManifest: TmsLayer[] = await Promise.all(
         servicesFromManifest
           .filter((tmsService) => tmsService.getId() === this._mapConfig.emsTileLayerId.bright)
-          .map(async (tmsService) => {
-            //shim for compatibility
+          .map(async (tmsService: TMSService) => {
+            // shim for compatibility
             return {
               origin: tmsService.getOrigin(),
               id: tmsService.getId(),
-              minZoom: await tmsService.getMinZoom(),
-              maxZoom: await tmsService.getMaxZoom(),
+              minZoom: (await tmsService.getMinZoom()) as number,
```

Comment:
tmsService.getMinZoom() can return number or undefined, https://github.com/elastic/ems-client/blob/master/src/tms_service.ts#L238, should we check for undefined before casting to number?

Reply:
imho I would not, because I'd be hesitant to tweak want thr existing logic. this  cast more of an artefact of the ts-conversion. We will be able to remove service_settings from VegaMaps, when we're starting to use on-prem, so then we can remove this class altogether.